src/frontend/apps/impress/src/features/docs/doc-tree/components/DocTree.tsx

```diff
@@ -0,0 +1,175 @@
+import { OpenMap, TreeView, TreeViewMoveResult } from '@gouvfr-lasuite/ui-kit';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import { css } from 'styled-components';
+
+import { Box, SeparatedSection, StyledLink } from '@/components';
+import { useCunninghamTheme } from '@/cunningham';
+
+import { Doc } from '../../doc-management';
+import { SimpleDocItem } from '../../docs-grid';
+import { useDocTree } from '../api/useDocTree';
+import { useMoveDoc } from '../api/useMove';
+import { useDocTreeData } from '../context/DocTreeContext';
+
+import { DocSubPageItem } from './DocSubPageItem';
+import { DocTreeItemActions } from './DocTreeItemActions';
+
+type DocTreeProps = {
+  initialTargetId: string;
+};
+export const DocTree = ({ initialTargetId }: DocTreeProps) => {
+  const { spacingsTokens } = useCunninghamTheme();
+  const spacing = spacingsTokens();
+  const treeData = useDocTreeData();
+  const router = useRouter();
+  const [initialOpenState, setInitialOpenState] = useState<OpenMap | undefined>(
+    undefined,
+  );
+
+  const { mutate: moveDoc } = useMoveDoc();
+
+  const { data } = useDocTree({
+    docId: initialTargetId,
+  });
+
+  const handleMove = (result: TreeViewMoveResult) => {
+    moveDoc({
+      sourceDocumentId: result.sourceId,
+      targetDocumentId: result.targetModeId,
+      position: result.mode,
+    });
+    treeData?.tree.handleMove(result);
+  };
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+    const { children: rootChildren, ...root } = data;
+    const children = rootChildren ?? [];
+    treeData?.setRoot(root);
+    const initialOpenState: OpenMap = {};
+    initialOpenState[root.id] = true;
+    const serialize = (children: Doc[]) => {
+      children.forEach((child) => {
+        child.childrenCount = child.numchild ?? 0;
+        if (child?.children?.length && child?.children?.length > 0) {
+          initialOpenState[child.id] = true;
+        }
+        serialize(child.children ?? []);
+      });
+    };
+    serialize(children);
+
+    treeData?.tree.resetTree(children);
+    setInitialOpenState(initialOpenState);
+    if (initialTargetId === root.id) {
+      treeData?.tree.setSelectedNode(root);
+    } else {
+      treeData?.tree.selectNodeById(initialTargetId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data]);
```

Comment:
Rarely good to omit dependencies, means often the pattern is not correct.
Can you describe by a comment why it is ommited.